compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -3358,12 +3361,15 @@ private BType findAssignableType(SymbolEnv env, BLangExpression selectExp, BType
         switch (collectionType.tag) {
             case TypeTags.STREAM:
                 BErrorType errorType = (BErrorType) ((BStreamType) collectionType).error;
-                if (!isStream) {
+                if (!isStream && !isTable) {
                     actualType = new BArrayType(actualType);
                     if (errorType != null) {
                         return BUnionType.create(null, actualType, errorType);
                     }
```

Comment:
Shall we move this to the else block? So it would be if isTable, else if isStream, and finally else?